Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIWindowsCocoa.mm

```diff
@@ -35,69 +35,501 @@
 #import "CocoaHelpers.h"
 #import "Logging.h"
 #import "MessageSenderInlines.h"
+#import "WebExtensionAPITabs.h"
 #import "WebExtensionAPIWindowsEvent.h"
 #import "WebExtensionContextMessages.h"
 #import "WebExtensionContextProxy.h"
+#import "WebExtensionWindowIdentifier.h"
 #import "WebProcess.h"
 #import "_WKWebExtensionUtilities.h"
-#import <wtf/DateMath.h>
+
+static NSString *populateKey = @"populate";
+
+static NSString *windowTypesKey = @"windowTypes";
+static NSString *normalKey = @"normal";
+static NSString *popupKey = @"popup";
+
+static NSString *minimizedKey = @"minimized";
+static NSString *maximizedKey = @"maximized";
+static NSString *fullscreenKey = @"fullscreen";
+
+static NSString *focusedKey = @"focused";
+static NSString *incognitoKey = @"incognito";
+static NSString *stateKey = @"state";
+static NSString *typeKey = @"type";
+
+static NSString *topKey = @"top";
+static NSString *leftKey = @"left";
+static NSString *widthKey = @"width";
+static NSString *heightKey = @"height";
+
+static NSString *tabsKey = @"tabs";
+
+static NSString *idKey = @"id";
+static NSString *alwaysOnTopKey = @"alwaysOnTop";
+
+static NSString *urlKey = @"url";
+static NSString *tabIdKey = @"tabId";
```

Comment:
Should these be `const`? Or does that not really matter here?

Reply:
Yeah they should be!

Reply:
That ended up causing build issues, since all uses need to be const and it was annoying.